src/parser/mod.rs

```diff
@@ -3571,6 +3572,11 @@ impl<'a> Parser<'a> {
                             ),
                             regexp,
                         })
+                    } else if dialect.supports_not_null() && negated && null {
+                        Ok(Expr::NotNull {
+                            expr: Box::new(expr),
+                            one_word: false,
```

Comment:
On second thought I do think I prefer inverting this to `with_space` instead of `one_word`, thoughts?